cmd/git-go-patch/rebase.go

```diff
@@ -50,5 +53,39 @@ func handleRebase(p subcmd.ParseFunc) error {
 	cmd.Stdin = os.Stdin
 	cmd.Dir = goDir
 
-	return executil.Run(cmd)
+	if err := executil.Run(cmd); err != nil {
+		return err
+	}
+
+	if err := warnIfOutsideSubmodule(goDir); err != nil {
+		// Just log for diagnosis, this warning isn't critical.
+		fmt.Printf("Unexpected error while checking wd: %v\n", err)
+	}
+	return nil
+}
+
+func warnIfOutsideSubmodule(submoduleDir string) error {
```

Comment:
I (slightly) prefer it this way because it lets me write ordinary error handling:

```go
	if err != nil {
		return fmt.Errorf("unable to get working dir: %v", wd)
	}

	if err != nil {
		fmt.Printf("Unexpected error while checking wd: unable to get working dir: %v\n", wd)
		return
	}
```

Generally speaking, it's idiomatic to have an `error` return most of the time, so to me, it's easier to read if the code leans into that.

I'll switch it over to see the difference for real. @qmuntal, also curious what your thoughts on this are.